### PR TITLE
Refactor documentation learn page

### PR DESCRIPTION
**Pull request by shehan360**
## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: swan-lake/learn/documenting-ballerina-code.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ redirect_from:
 
 Developers can write the documentation inline with the Ballerina source code using the lightweight [markdown](https://daringfireball.net/projects/markdown/syntax) markup language. They can document special constructs such as parameters, return values, fields, etc. within the code using documentation attributes. Once the code is documented, developers can generate a basic HTML version of their Ballerina modules using the `ballerina doc` command. Developers are encouraged to have their custom themes and styles, to have a standard presentation of their Ballerina documentation.
 
-Ballerina documentation design and usage is aligned with project and module semantics of Ballerina. You can generate documentation for modules using the `ballerina doc` command.
+Ballerina documentation design and usage is aligned with package and module semantics of Ballerina. You can generate documentation for modules using the `ballerina doc` command.
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:19]:
```suggestion
Ballerina documentation design and usage is aligned with the package and module semantics of Ballerina. You can generate documentation for modules using the `ballerina doc` command.
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 2]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
 
-A typical project structure of a Ballerina project is like this:
+A typical package structure of a Ballerina package is like this:
 
 ```
 /
-  Ballerina.toml       # Configuration that defines project intent
-    src
-      module1/             # The source in this directory will be named “<org-name>/module1”
+  Ballerina.toml       # Configuration that defines package intent
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:102]:
```suggestion
  Ballerina.toml       # Configuration, which defines the package intent.
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 3]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
+  main.bal
+  Module.md            # Optional, contains descriptive metadata of the default module for display at
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:104]:
```suggestion
  Module.md            # Contains descriptive metadata of the default module to be displayed in
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 4]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
+                       # Ballerina Central
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:105]:
```suggestion
                       # Ballerina Central. This is optional.
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 5]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
 ```
 /
-  Ballerina.toml       # Configuration that defines project intent
-    src
-      module1/             # The source in this directory will be named “<org-name>/module1”
+  Ballerina.toml       # Configuration that defines package intent
+  main.bal
+  Module.md            # Optional, contains descriptive metadata of the default module for display at
+                       # Ballerina Central        
+  Package.md           # Optional, contains descriptive metadata of the package for display at
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:106]:
```suggestion
  Package.md           # Contains descriptive metadata of the package to be displayed in
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 6]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
+                       # Ballerina Central
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:107]:
```suggestion
                       # Ballerina Central. This is optional.
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 7]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
-      module1/             # The source in this directory will be named “<org-name>/module1”
+  Ballerina.toml       # Configuration that defines package intent
+  main.bal
+  Module.md            # Optional, contains descriptive metadata of the default module for display at
+                       # Ballerina Central        
+  Package.md           # Optional, contains descriptive metadata of the package for display at
+                       # Ballerina Central  
+    modules
+      module1/             
         Module.md          # Optional, contains descriptive metadata for display at
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:110]:
```suggestion
        Module.md          # Contains descriptive metadata to be displayed in
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 8]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
                            # Ballerina Central
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:111]:
```suggestion
                           # Ballerina Central. This is optional.
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 9]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
 
-      modules.can.include.dots.in.dir.name/
+      module2/
         Module.md
         *.bal
         [tests/]
         [resources/]
 
-  [resources/]         # Resources included with every module in the project
+  [resources/]         # Resources included with every module in the package
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:122]:
```suggestion
  [resources/]         # Resources included in every module in the package.
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 10]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
 
   target/              # Compiled executables and other artifacts end up here
 ```
 
-The `ballerina doc` command will read the `Module.md` and prepend it to the generated HTML file.
+The `ballerina doc` command will read the `Package.md` and `Module.md` and prepend it to the generated HTML file.
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:127]:
```suggestion
The `ballerina doc` command will read the `Package.md` and `Module.md` files and prepend it to the generated HTML file.
```
What is "it" here?

File: swan-lake/learn/documenting-ballerina-code.md [hunk 11]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
 
 Check [HTTP module documentation](/swan-lake/learn/api-docs/ballerina/http/index.html) for sample HTML that has `Module.md` content at the top, followed by the other module constructs.
 
 
 ## Generating Ballerina Documentation
 
-Ballerina provides a `doc` command, which can be executed against a given Ballerina project. This command will result in generating the Ballerina documentation as HTML files, for all the modules in the project.
+Ballerina provides a `doc` command, which can be executed against a given Ballerina package. This command will result in generating the Ballerina documentation as HTML files, for all the modules in the package.
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:134]:
```suggestion
Ballerina provides a `doc` command, which can be executed against a given Ballerina package. This command will result in generating the Ballerina documentation as HTML files for all the modules in the package.
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 12]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
 
-First, let's create a new Ballerina project:
+First, let's create a new Ballerina package:
 
 ```bash
-$ ballerina new myproject
-Created new Ballerina project at myproject
+$ ballerina new math
+Created new Ballerina package 'math' at math
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:140]:
```suggestion
Created new Ballerina package 'math' at math
```
What is the second "math" here?

File: swan-lake/learn/documenting-ballerina-code.md [hunk 13]
```diff
@@ -95,56 +95,68 @@ public function submit(@sensitive string httpVerb, string path, Request request)
...
+Next, move into the package directory and execute `ballerina add <module-name>` to add a new Ballerina module.
 
 ```bash
-$ cd myproject/
-$ ballerina add math
-Added new ballerina module at 'src/math'
+$ cd math/
 $ ballerina add world
-Added new ballerina module at 'src/world'
+Added new ballerina module at 'modules/world'
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:148]:
```suggestion
Added new ballerina module at 'modules/world'.
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 14]
```diff
@@ -196,34 +208,24 @@ public class Person {
...
-        foo/world:0.1.0
-
+	user/math:0.1.0
 Generating API Documentation
-        target/apidocs
+Saved to: target/apidocs
 ```
 
-`target/apidocs/` folder would contain following;
+`target/apidocs/math` folder would contain following;
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:223]:
```suggestion
`target/apidocs/math` folder would contain the following;
```

File: swan-lake/learn/documenting-ballerina-code.md [hunk 15]
```diff
@@ -196,34 +208,24 @@ public class Person {
...
-* `index.html`  - contains an index page of all the modules in the Ballerina project 
-* `math` - contains the documentation of the module named `math`
-* `world` - contains the documentation of the module named `world`
-
-If you want to generate documentation for a selected Ballerina module, then you can execute the following command from the Ballerina project root directory:
-
-```bash
-$ ballerina doc <module_name>
-```
+* `index.html`  - the entry point for documentation.
```
Comment (praneesha) [inline swan-lake/learn/documenting-ballerina-code.md:229]:
```suggestion
* `index.html`  - the entry point for documentation
```

File: swan-lake/learn/documenting-ballerina-code.md (hunk 12 above)
Comment (shehan360) [reply swan-lake/learn/documenting-ballerina-code.md:140]: This is the console output from ballerina new command

Comment (praneesha) [reply swan-lake/learn/documenting-ballerina-code.md:140]: Ack. Fine.

**Review by praneesha — APPROVED** [review]

**Pull request merged**